packages/vite-plugin-svelte/src/index.ts

```diff
@@ -154,21 +154,29 @@ export function svelte(inlineOptions?: Partial<Options>): Plugin[] {
 					}
 					return resolvedSvelteSSR;
 				}
-				try {
-					const resolved = resolveViaPackageJsonSvelte(importee, importer, cache);
-					if (resolved) {
-						log.debug(
-							`resolveId resolved ${resolved} via package.json svelte field of ${importee}`
+				//@ts-expect-error scan
+				const scan = !!opts?.scan; // scanner phase of optimizeDeps
+				const isPrebundled =
+					options.prebundleSvelteLibraries && viteConfig.optimizeDeps?.include?.includes(importee);
+				// for prebundled libraries we let vite resolve the prebundling result
+				// for ssr, during scanning and non-prebundled, we do it
+				if (ssr || scan || !isPrebundled) {
```

Comment:
If there is an `exports` then you potentially may resolve to a very different place during the scan and non-scan phases, which doesn't seem right to me?

Reply:
no, thats excactly what we want. scan should find the package as described by the svelte field, optimize it and then it lives in `.vite/deps/something.js`  so after scan we want vite itself to resolve to that prebundled version.

Reply:
Imagine the Svelte field points to "index.svelte" and the `exports` map has nothing but `package.json`. Then I think Vite would fail to resolve anything after scan?

Reply:
if the scan returned the .svelte file, the optimizer should have created a bundle in /node_modules/.vite for it. but now that i write this, thats not used during build. So we would additionally have to do it on build too unless optimize for build is set?

Reply:
Ah yeah we should probably handle build differently by checking `optimizeDeps.disabled` too. By default, build should always resolve `pkg.svelte` regardless of prebundling.

Reply:
done